agents.md

```diff
@@ -0,0 +1,501 @@
+
+### Build Process
+
+Multi-stage Docker build:
+
+1. Node stage: Build frontend static files
+2. Python stage: Install dependencies, collect static files
+3. Django serves via Whitenoise + Gunicorn
+
+### Monitoring
+
+- Metrics: markus with datadog extensions → telegraf → influxdb
+- Errors: Sentry
+- Profiling: Google Cloud Profiler
+- Telemetry: Glean
+- Logs: MozLog format
+
+Enable metrics in development:
+
+- `STATSD_ENABLED=True` - Send to statsd server (use `nc -lu localhost 8125` to view)
+- `STATSD_DEBUG=True` - Log metrics to console
+
+Utility functions in `emails/utils.py`:
+
+- `time_if_enabled(name)`
+- `incr_if_enabled(name, value=1, tags=None)`
+- `histogram_if_enabled(name, value, tags=None)`
+- `gauge_if_enabled(name, value, tags=None)`
+
+## Key Files
+
+- `privaterelay/settings.py` - All Django configuration
+- `api/views/emails.py` - Email API endpoints
+- `api/views/phones.py` - Phone API endpoints
+- `api/views/privaterelay.py` - Account/profile API
+- `emails/models.py` - Email masking models and business logic
+- `emails/views.py` - Email forwarding handlers
+- `frontend/src/pages/` - Next.js page routes
+- `frontend/src/hooks/api/` - SWR data fetching hooks
+- `docs/` - Comprehensive documentation
+
+## Technology Stack
+
+- **Backend**: Python 3, Django, DRF, Gunicorn
+- **Frontend**: TypeScript, React, Next.js
+- **Database**: PostgreSQL
+- **Caching**: Redis via django-redis
+- **Email**: AWS SES/SNS via boto3
+- **Phone**: Twilio
+- **Auth**: Mozilla Accounts OAuth via django-allauth
+- **Testing**: pytest-django, Jest, Playwright, React Testing Library
+- **Code Quality**: Ruff, mypy, ESLint, Prettier, StyleLint
+
+## Code Quality Guidelines
+
+When writing code for this project, follow these principles to keep it clean and maintainable.
+
+### Write Concise Code
+
+- Use clear variable names. Avoid comments that restate the code.
+- Prefer built-in functions over custom implementations.
+- Remove debug code before committing (console.log, print statements, etc.).
+
+**Example**:
+
+```python
+# ❌ Verbose
+def get_user_data(user_id):
+    # Get the user from the database
+    user = db.query(User).filter(User.id == user_id).first()
+    # Return the user
+    return user
+
+# ✅ Concise
+def get_user(user_id):
+    return db.query(User).filter_by(id=user_id).first()
+```
+
+### When to Add Comments
+
+Add comments only for:
+
+- **Why, not what** - Explain reasoning, not mechanics
+- **Context** - Business rules, edge cases, non-obvious decisions
+- **Warnings** - Performance gotchas, security considerations
+
+Skip comments that:
+
+- Restate what the code already says
+- Describe obvious operations
+- Would be better expressed as better variable names
+
+### When to Extract Functions
+
+**Extract a function when:**
+
+- The same code appears 3+ times (not 2)
+- A block genuinely improves readability when named
+- It needs independent testing
```

Comment:
Nit: @groovecoder do you still prefer to extract functions if it indents too many levels?

Reply:
Hmm ... good question. What do you think? Seems like a decent point to add.

Reply:
I added it as "In python code, when it is indented too many levels"